rmw_connext_dynamic_cpp/src/functions.cpp

```diff
@@ -2261,11 +2171,11 @@ rmw_create_client(
 
   // create requester
   {
-    if (!get_datareader_qos(participant, datareader_qos)) {
+    if (!get_datareader_qos(qos_profile, participant, datareader_qos)) {
```

Comment:
I would pass `qos_profile` as a second or third argument here (as well as below). Not really sure why - most other functions take the participant as a first argument.